content/post/2022-02-21-code-style/index.Rmd

```diff
@@ -0,0 +1,173 @@
+---
+slug: code-style
+title: "Coding style, coding etiquette" 
+authors: 
+- Maëlle Salmon 
+- Christophe Dervieux
+date: "2022-02-21" 
+tags: 
+- package development 
+output: hugodown::hugo_document
+---
+
+```{r setup, include=FALSE}
+
+knitr::opts_chunk$set(fig.path = "", comment = "")
+# knitr hook to make images output use Hugo options
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{{<figure src=",
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
+        glue::glue_collapse(
+          glue::glue('{names(hugoopts)}="{hugoopts}"'),
+          sep = " "
+        )
+      },
+      ">}}\n"
+    )
+  }
+)
+
+```
+
+Do you indent your code with [one tab, two spaces](https://www.youtube.com/watch?v=SsoOG6ZeyUI), or eight spaces?
+Do you feel strongly about the location of the curly brace closing a function definition?
+Do you have naming preferences?
+You probably have picked up some habits along the way.
+In any case, having some sort of consistency in coding style will help those who read the code to understand, fix or enhance it.
+In this post, we shall share some resources about coding style, useful tools, and some remarks on etiquette.
+
+## What *is* coding style?
+
+Coding style is a set of rules about well, aesthetics (aligning and code spacing) but also naming (of variables, functions, etc.), commenting, structuring (e.g. avoiding complex logic), etc.
+These rules help with better code clarity and collaboration.
+
+Sometimes somes rules are enforced by the language (indentation in python), sometimes the language is quite loose, like... R where you can add a lot of spaces, and so in that case more style guides exist.
+
+Coding style has an universal goal: making your code easier to understand and maintain, for you later and for your code collaborator.
+Having a team/company style guide is common practice.
+Major Tech Companies have style guides, e.g.
+[Google](https://google.github.io/styleguide/)[^1].
+
+[^1]: Note that the Google R style guide inspired the tidyverse styleguide but Google now refers to the tidyverse style guide as R style guide.
+
+However there is no "right" coding style as the "correct" choice depends on personal preferences, and the [constraints at hand](https://www.heltweg.org/posts/who-wrote-this-shit/).
+
+## Resources
+
+One way to develop a sense of coding style is to read a lot of code, but where to learn about coding style explicitly?
+
+### Specific R resources
+
+The most popular, or at least most documented style guide out there is the [tidyverse style guide](https://style.tidyverse.org).
+
+Some variations on it include
```

Comment:
At first I did not catch that, but should we really use the word variation ? It sounds like it is derived and at least some of the rules follow. I see the below list as other different organisation or personal preferences and not specifically a variation of tidyverse one. 

I think we should present it as tidyverse one is the norm and other derived from it. 

```suggestion
Some other style preferences include
```